pkg/inference/discover_test.go

```diff
@@ -100,3 +103,19 @@ func TestDiscover(t *testing.T) {
 		})
 	})
 }
+
+func TestDiscoverMarshalling(t *testing.T) {
+	testCase(t, func(c *testContext) {
+		Register(&TestProvider{Name: "provider-one", Distant: false, Available: true})
+		Register(&TestProvider{Name: "provider-two", Distant: false, Available: true})
+		inferences := Discover(config.New())
+		bytes, err := json.Marshal(inferences)
+		t.Run("Marshalling returns no error", func(t *testing.T) {
+			assert.Nil(t, err, "expected no error when marshalling inferences")
+		})
+		t.Run("Marshalling returns expected JSON", func(t *testing.T) {
+			assert.JSONEq(t, `[{"name":"provider-one","Distant":false},{"name":"provider-two","Distant":false}]`, string(bytes),
```

Comment:
You may add a ``json:"distant"` to the structure `Attributes.Distant` field, to get a `distant` field in json

Reply:
I didn't work on this field because I'm not sure we want to rename it to remote or local.
But I can add the json tag for now.